### PR TITLE
 #150704024  front end tests

### PR DESCRIPTION
**Pull request by ogomb**
#### What does this PR do?
Tests the front-end part of the application

#### How should this be manually tested?
run this command `python manage.py test hc/front/tests`

#### What are the relevant pivotal tracker stories?
#150704024

### CODE REVIEW

File: hc/front/tests/test_add_channel.py [hunk 1]
```diff
@@ -37,5 +37,24 @@ def test_instructions_work(self):
             r = self.client.get(url)
             self.assertContains(r, "Integration Settings", status_code=200)
 
+    def test_bad_kinds_not_working(self):
+        self.client.login(username="alice@example.org", password="password")
+        bad_kind = "twitter"
+        url = "/integrations/add_%s/" % bad_kind
+        r = self.client.get(url)
```
Comment (collinmutembei) [inline hc/front/tests/test_add_channel.py]: Use a more descriptive variable name, in place of `r`. See [here](http://pep8.org/#names-to-avoid)

File: hc/front/tests/test_add_channel.py [hunk 2]
```diff
@@ -37,5 +37,24 @@ def test_instructions_work(self):
...
+        self.assertEqual(r.status_code, 404)
+         
     ### Test that the team access works
+    def test_can_add_channel_using_team_access(self):
+        url = "/integrations/add/"
+        form = {"kind": "email", "value": "bob@example.org"}
+
+        # Bob is in alice's team so he should be able to add a channel
+        self.client.login(username="bob@example.org", password="password")
+        r = self.client.post(url, form)
```
Comment (collinmutembei) [inline hc/front/tests/test_add_channel.py]: Adding an integration does not sufficiently check that user registered with the email *bob@example.org* has  access to Alice's team. Use a different approach.

File: hc/front/tests/test_add_channel.py (hunk 1 above)
Comment (collinmutembei) [reply hc/front/tests/test_add_channel.py]: And any other line you have similar variables. Like this [one](https://github.com/andela/healthchecks-f-society/blob/0a00ed2b0eb44ad02e2d4bea785ec04bce80ca79/hc/front/tests/test_add_pushbover.py#L56)

**Review by collinmutembei — CHANGES_REQUESTED** [review]
Attend to the review comments I have added @ogomb @briankabiro

File: hc/front/tests/test_add_channel.py [hunk 3]
```diff
@@ -43,13 +43,18 @@ def test_bad_kinds_not_working(self):
...
-
+         
+    ### Test that the team access works
+    def test_can_add_channel_using_team_access(self):
+        url = "/integrations/add/"
+        form = {"kind": "email", "value": "bob@example.org"}
 
+        # Bob is in alice's team so he should be able to add a channel
+        self.client.login(username="bob@example.org", password="password")
+        r = self.client.post(url, form)
```
Comment (briankabiro) [inline hc/front/tests/test_add_channel.py]: hey @collinmutembei Could that be addressed by the assert I made below, by confirming that the integration belongs to alice?

Comment (collinmutembei) [reply hc/front/tests/test_add_channel.py]: @briankabiro Does a channel have a relation with an integration?

Comment (collinmutembei) [reply hc/front/tests/test_add_channel.py]: 🤔 Come to think of it; it makes sense @briankabiro

File: hc/front/tests/test_add_pushbover.py [hunk 4]
```diff
@@ -38,3 +42,55 @@ def test_it_validates_nonce(self):
...
+            username='alice@example.org', password='password')
+
+        session = self.client.session
+        # set po_once to n
+        session["po_nonce"] = "n"
+        session.save()
+        # set prio to a number greater than 2
+        params = "pushover_user_key=a&nonce=n&prio=3"
+        response = self.client.get('/integrations/add_pushover/?%s' % params)
+        self.assertEqual(r.status_code, 400)
```
Comment (collinmutembei) [inline hc/front/tests/test_add_pushbover.py]: Update `r` to  `response` on [line 57](https://github.com/andela/healthchecks-f-society/blob/de13179751837d79159219873706b4757da6e5ae/hc/front/tests/test_add_pushbover.py#L57) to ensure that the build passes.

**Review by collinmutembei — CHANGES_REQUESTED** [review]
Fix the failing build, otherwise :shipit:

**Comment by briankabiro** [comment]
@collinmutembei On it right now

**Review by collinmutembei — APPROVED** [review]

**Pull request merged**